packages/@n8n/db/src/migrations/common/1765361177092-AddIconToAgentTable.ts

```diff
@@ -0,0 +1,19 @@
+import type { MigrationContext, ReversibleMigration } from '../migration-types';
+
+const agentsTableName = 'chat_hub_agents';
+const sessionsTableName = 'chat_hub_sessions';
+
+export class AddIconToAgentTable1765361177092 implements ReversibleMigration {
+	async up({ schemaBuilder: { addColumns, column } }: MigrationContext) {
+		// Add icon column to agents table (nullable)
+		await addColumns(agentsTableName, [column('icon').json]);
+
+		// Add agentIcon column to sessions table (nullable)
+		await addColumns(sessionsTableName, [column('agentIcon').json]);
```

Comment:
I'm questioning duplicating this field on the sessions table a bit - while I get that this can be handy can we just do a join?

Reply:
For personal agents it should be straightforward, for workflow agents it's more tricky because the icon is actually in the project table. But we can also make another query and merge data in service.

Reply:
"Just do a join" worked. Well, not a join but some joins. And it also solves the problem of icons in sessions table get out of sync when it is changed after the conversation. Thank you for the suggestion. [a061d6b](https://github.com/n8n-io/n8n/pull/23032/commits/a061d6b890a700e9492e605cb8d91b292aef7c71)